src/util/invariant.h

```diff
@@ -117,20 +116,24 @@ class invariant_failedt: public std::logic_error
 #define INVARIANT(CONDITION, REASON) \
   __CPROVER_assert((CONDITION), "Invariant : " REASON)
 
+#define INVARIANT_STRUCTURED(CONDITION, TYPENAME, ...) \
+  INVARIANT(CONDITION, "")
 
 #elif defined(CPROVER_INVARIANT_DO_NOT_CHECK)
 // For performance builds, invariants can be ignored
 // This is *not* recommended as it can result in unpredictable behaviour
 // including silently reporting incorrect results.
 // This is also useful for checking side-effect freedom.
-#define INVARIANT(CONDITION, REASON, ...) do {} while(0)
+#define INVARIANT(CONDITION, REASON) do {} while(0)
+#define INVARIANT_STRUCTURED(CONDITION, TYPENAME, ...) do {} while(0)
 
```

Comment:
Having the ellipsis in the macro definition results in a warning, and without defining INVARIANT_STRUCTURED compilation will always fails. How did this ever work?

Reply:
Which warning would that be? It would be surprising that you are the first one to observe it?!

Reply:
I was surprised as well. See the discussion of the [first commit](https://github.com/diffblue/cbmc/pull/1477/commits/56186c6cbf6debf1516dab4127da04702dac6166).

Reply:
I suspect it didn't.  The structured invariant stuff was added after the initial invariant support and it may have simply not been tested.  I don't recall reviewing it.